src/pricing-fetcher.ts

```
@@ -26,14 +27,42 @@ export class PricingFetcher implements Disposable {
+		const httpsProxy = process.env.HTTPS_PROXY ?? process.env.https_proxy ?? '';
+		const httpProxy = process.env.HTTP_PROXY ?? process.env.http_proxy ?? '';
+		const proxyUrl = httpsProxy.length > 0 ? httpsProxy : httpProxy;
+
+		if (proxyUrl.length === 0) {
+			return {};
+		}
+
+		// Use undici's ProxyAgent for proper proxy support
+		try {
+			// eslint-disable-next-line ts/no-require-imports
+			const { ProxyAgent } = require('undici') as typeof import('undici');
```

ESLint

[error] Unsafe assignment of an error typed value.

(ts/no-unsafe-assignment)